### DISCUSSION TITLE
Clean up dependencies and maybe use PackageExtension

### DISCUSSION THREAD

**Discussion by SimonDanisch** [JuliaPlots/Makie.jl#4247] [Ideas]
#### Good candidate for PackageExtension, since user needs to load the package to use the feature
* SparseArrays 
* OffsetArrays

#### Not sure how to use PackageExtension but packages feel optional
* InteractiveUtils
* Isoband
* KernelDensity
* StatsFuns 
* PolygonOps
* TriplotBase 

#### Can be removed completely with little effort
- [ ] StructArrays 
- [x] UnicodeFun 
- [x] Match
- [x] Setfield

#### Needed only for texture atlas (maybe nice to move out for CairoMakie?)

* SignedDistanceFields 
* StableHashTraits 
* Packing 

We should also check if all `__init__` functions are fast for all dependencies and Makie itself.

**Comment by asinghvi17** [comment]
~One potential issue with package extensions is that functions need to initially be defined in the main module, because you can't dynamically export or easily access functions defined in extension modules (you have to load the module using e.g. `Base.get_extension(Makie, :KernelDensityExt)`).~

We could define the recipes in the main module, and the `plot!` implementations in the extension modules, which should work pretty easily.  It would make usability a little worse, though, since naively using `density` wouldn't work.

**Comment by bilderbuchi** [comment]
Related: #2505 (additional candidate for PackageExtension: `FFMPEG`?)

> It would make usability a little worse, though, since naively using `density` wouldn't work.

Could you emit an actionable error message for users hitting that situation? `Please install package XYZ to use this feature`?

**Comment by bilderbuchi** [comment]
> Related: #2505 (additional candidate for PackageExtension: `FFMPEG`?)
> 
> > It would make usability a little worse, though, since naively using `density` wouldn't work.
> 
> Could you emit an actionable error message for users hitting that situation? `Please install package XYZ to use this feature`?

Apparently such a technique is now available (from https://github.com/MakieOrg/Makie.jl/issues/3953#issuecomment-2267145860):
> users who try eg `qqplot()` will see an error saying "please execute using Distributions to use qqplot()". It clear and unambiguous what exactly to do.
